internal/graph/graph.go

```diff
@@ -442,12 +444,32 @@ func ShortenFunctionName(f string) string {
 	f = cppAnonymousPrefixRegExp.ReplaceAllString(f, "")
 	for _, re := range []*regexp.Regexp{goRegExp, javaRegExp, cppRegExp} {
 		if matches := re.FindStringSubmatch(f); len(matches) >= 2 {
-			return strings.Join(matches[1:], "")
+			name := strings.Join(matches[1:], "")
+			if re == goRegExp {
+				return shortenGoFunc(f, name)
```

Comment:
I feel it might be simpler to first remove the version substring from the name, and then handle it just like before.

Reply:
Not sure if you saw the previous review comments, but if preferred this all can be removed and replaced with a single regex change.

Reply:
> I feel it might be simpler to first remove the version substring from the name, and then handle it just like before.

I'm not sure how this is possible; the name here is extracted from the regex directly. If we remove the version suffix, you get the empty string.

Reply:
Oh, I see, you mean that if the name matches a version, remove the suffix and then try again. It wouldn't distinguish two versions of the same module, but I guess it's no worse than any other name aliasing within the same graph.

Reply:
I think I misinterpreted again (sorry!), so I'll wait for clarification.

I think you meant just starting with something like `github.com/jackc/pgx/v4/foo.bar`, then replacing the first instance of `/v4/` with `/`, then running the regex again. Not quite a suffix, but functional enough.

Reply:
Yes, I meant first remove all occurrences of `/[1-9][0-9]*[./]`, then business as usual.

Reply:
Sorry, `/v[1-9][0-9]*[./]`.

Reply:
And not all occurrences but at most one occurrence (assuming there can't be two version substrings in the name).

Reply:
All instances wouldn't work, because it's legal for me to write `github.com/foo/bar/v4/something/v8` or similar. First instance I believe would work as intended, though.

Reply:
I'll give it a try. Would be `v([2-9]|[1-9][0-9]+)\.`, though, as `v0` and `v1` don't exist.

Reply:
Done, and retitled to match the new fix. `regexp` doesn't have a nice "just replace once", so I used the replace all with two capture groups method.